lib/rouge/lexer.rb

```diff
@@ -405,15 +413,22 @@ def reset!
 
     # Given a string, yield [token, chunk] pairs.  If no block is given,
     # an enumerator is returned.
-    #
-    # @option opts :continue
-    #   Continue the lex from the previous state (i.e. don't call #reset!)
-    def lex(string, opts={}, &b)
-      return enum_for(:lex, string, opts) unless block_given?
+    def lex(string, opts=nil, &b)
+      if opts
+        warn 'the :continue option to Formatter#lex is deprecated, use #continue_lex instead.'
+      end
+
+      return enum_for(:lex, string) unless block_given?
 
       Lexer.assert_utf8!(string)
+      reset!
```

Comment:
To confirm: if a user calls `#lex` as they were doing previously, the call won't raise an error but: (a) it will issue the warning you've added above; and (b) it will not work as expected given the call here to `reset!`. This is still a breaking change, then?

Reply:
I think having the following would suffice:
```ruby
if opts
  warn deprecation_msg
  old_behavior
else
  new_behavior
end
```
Optimizations being available to only those that make the switch to the new behavior..

Reply:
Oops, my bad - I meant to put a call to `continue_lex` in there